src/builders/indexable-link-builder.php

```diff
@@ -84,7 +84,12 @@ public function set_dependencies(
 	 */
 	public function build( $indexable, $content ) {
 		if ( $indexable->object_type === 'post' ) {
+			$post = \get_post( $indexable->object_id );
+			// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited -- To setup the post we need to do this explicitly.
+			$GLOBALS['post'] = $post;
+			setup_postdata( $post );
 			$content = \apply_filters( 'the_content', $content );
+			wp_reset_postdata();
```

Comment:
elsewhere in our codebase I see the  comment below; would that apply here as well?

// phpcs:ignore WordPress.WP.DiscouragedFunctions.wp_reset_query_wp_reset_query -- Reason: The recommended function, wp_reset_postdata, doesn't reset wp_query.
--
249 | \wp_reset_query();

Reply:
In this case it's specifically the post data we're interested in as we're not manipulating the query itself.

We're just making sure the global WordPress expects are set when we apply `the_content` filter, we don't touch the `$wp_query` global.